### PR TITLE
refactor: consistently use `binaryTarget` term

### PR DESCRIPTION
**Pull request by aqrln**
Right now we use the term "platform" in different contexts to signify
different things, and use either "platform" or "binaryTarget" in the
same context to mean the same thing. This commit consolidates the usage
of the latter to just "binaryTarget" as it matches the user-facing API
in the form of the `binaryTargets` property in Prisma schema. Since it
is less ambiguous, it is easy to search and replace it later if we
decide on a different name.

Closes: https://github.com/prisma/team-orm/issues/366

/integration

### CODE REVIEW

File: packages/cli/src/Version.ts
Comment (janpio) [inline packages/cli/src/Version.ts:2]: package as well maybe?

**Comment by Jolg42** [comment]
@aqrln See failing tests, it looks like this breaks Studio

File: packages/cli/src/Version.ts
Comment (Jolg42) [reply packages/cli/src/Version.ts:2]: Let's keep this out of scope of this PR for now

**Comment by Jolg42** [comment]
@aqrln ⚠️ that the rename of `getPlatform` will be breaking for some other things I found:

- I'm not sure `reflector` is still used by PDP at all, though https://github.com/prisma/reflector/blob/b40cf4e660372a680558d1fb6f9aa6b7d3d4d617/src/Engines.ts#L4
- https://github.com/prisma/mini-proxy
- https://github.com/odroe/prisma-dmmf
- https://github.com/michalbundyra/prisma-distroless
- and a couple of archived repos

See search https://github.com/search?q=getPlatform+%22%40prisma%2Fget-platform%22+-repo%3Aprisma%2Fprisma+-is%3Afork++-language%3AJSON+-language%3AMarkdown&type=code

It looks minor overall, `mini-proxy` will definitely need to be updated and for the other public repos from the community they will have to rename on their own.

**Comment by jkomyno** [comment]
> @aqrln ⚠️ that the rename of `getPlatform` will be breaking for some other things I found:
> 
> * I'm not sure `reflector` is still used by PDP at all, though https://github.com/prisma/reflector/blob/b40cf4e660372a680558d1fb6f9aa6b7d3d4d617/src/Engines.ts#L4
> * https://github.com/prisma/mini-proxy
> * https://github.com/odroe/prisma-dmmf
> * https://github.com/michalbundyra/prisma-distroless
> * and a couple of archived repos
> 
> See search https://github.com/search?q=getPlatform+%22%40prisma%2Fget-platform%22+-repo%3Aprisma%2Fprisma+-is%3Afork++-language%3AJSON+-language%3AMarkdown&type=code
> 
> It looks minor overall, `mini-proxy` will definitely need to be updated and for the other public repos from the community they will have to rename on their own.

Are we ok with merging this PR, then? Will we reach out to the maintainers of these repos during the next sprint?

P.s.: this PR currently has conflicts, can you please fix them @aqrln? Thanks!

File: packages/client/src/runtime/core/engines/library/LibraryEngine.ts
Comment (Jolg42) [inline packages/client/src/runtime/core/engines/library/LibraryEngine.ts:324]: Looks like the property here could be renamed too? Possibly in a separate PR

**Review by Jolg42 — APPROVED** [review]
Changes look great to me ✨ 

See small comments

I asked Jason about reflector in https://prisma-company.slack.com/archives/CUXLS0Z6K/p1699979729090339

**Comment by Jolg42** [comment]
Let's see tests results after the sync

**Comment by aqrln** [comment]
Needs https://github.com/prisma/studio-code/pull/527 to be merged first (or integration releases to be fixed in studio) because otherwise it doesn't even build otherwise with the latest changes.

**Comment by aqrln** [comment]
mini-proxy adjusted in https://github.com/prisma/mini-proxy/pull/53 but it's not really breaking it because it does not depend on the version of `@prisma/internals` from the monorepo, it has its own bundled version of `@prisma/internals` (4.5.0 at the moment). The code path it is used on is a remnant of the initial implementation and is not even used by us in CI.

File: packages/internals/src/errorReporting.ts
Comment (Jolg42) [inline packages/internals/src/errorReporting.ts:20]:
```suggestion
  // We wanted to rename this to `binaryTarget` but we had to skip the rename here
  // because we didn't want to create more work
  // as the error handling backend would need to be updated and re-deployed
  // https://github.com/prisma/error-handling-backend
  platform: string
```

File: packages/client/src/generation/utils/buildNFTAnnotations.ts
Comment (Jolg42) [inline packages/client/src/generation/utils/buildNFTAnnotations.ts:14]:
```suggestion
 * @param binaryTargets the targeted binaryTargets
```
Weird but more accurate?

**Review by Jolg42 — APPROVED** [review]
That's a lot of changes 😄 

Nice and clean ✨  ✅

**Comment by aqrln** [comment]
reapplied a commit with review suggestions that got lost after rebase (didn't pull the changes before force pushing)

**Comment by aqrln** [comment]
Next steps:
* wait for the dev release
* bump prisma version in studio again
* bump studio version in prisma again

**Pull request merged**
